File: _group_anagrams.ts
function groupAnagrams(strs: string[]): string[][] {

    let map = new Map();
    let res = []

    for(let i = 0;i<strs.length;i++){
        let temp = strs[i].split('').sort().join('');
        if(map.has(temp)){
            let val = map.get(temp);
            val.push(strs[i]);
        }
        else{
            map.set(temp,[strs[i]]);
        }
    }

    for(let j of map)
    {
        res.push(j[1])
    }
    return res;

};

File: Python/Group_anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        dict = {}
        res = []

        for s in strs:
            v = ''.join(sorted(s))

            if( v in dict.keys()):
                dict[v].append(s)
            else:
                dict[v] = []
                dict[v].append(s)
        
        for i in dict.values():
            res.append(i)
        
        return res
            
